services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/taydennys/TaydennysAttachmentContentService.kt

```diff
@@ -0,0 +1,49 @@
+package fi.hel.haitaton.hanke.attachment.taydennys
+
+import fi.hel.haitaton.hanke.attachment.azure.Container
+import fi.hel.haitaton.hanke.attachment.common.FileClient
+import java.util.UUID
+import mu.KotlinLogging
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class TaydennysAttachmentContentService(val fileClient: FileClient) {
+    fun upload(
+        filename: String,
+        contentType: MediaType,
+        content: ByteArray,
+        taydennysId: UUID,
+    ): String {
+        val blobPath = generateBlobPath(taydennysId)
+        fileClient.upload(Container.HAKEMUS_LIITTEET, blobPath, filename, contentType, content)
+        logger.info { "Attachment content saved to $blobPath" }
+        return blobPath
+    }
+
+    fun delete(blobPath: String): Boolean =
+        fileClient.delete(Container.HAKEMUS_LIITTEET, blobPath).also {
+            if (it) {
+                logger.info { "Attachment content at $blobPath deleted" }
+            } else {
+                logger.warn { "Attachment content at $blobPath not found" }
+            }
+        }
+
+    companion object {
+        fun generateBlobPath(taydennysId: UUID) =
+            "${prefix(taydennysId)}${UUID.randomUUID()}"
+                .also { logger.info { "Generated blob path: $it" } }
+
+        /**
+         * Prefix derived from täydennys each attachment of that täydennys should have in the cloud
+         * storage.
+         *
+         * Used to distinguish the attachments of different täydennys from each other in the cloud
+         * storage and to enable deleting all of them at once.
+         */
+        fun prefix(taydennysId: UUID): String = "$taydennysId/"
```

Comment:
The [specs](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HAI/pages/9033580626/T+ydennyspyynt#POST-%2Ftaydennykset%2F%3Aid%2Fliitteet) say that these attachments should be in the same folder as the attachments of the hakemus. So that when the hakemus is removed, all the attachments related to it are removed as well.

If the folder is for the täydennys, we would need to add logic for deleting the folder to the täydennys removal, but it can also happen when the hakemus or the täydennyspyyntö is removed, so it would get more complicated.

Reply:
It could had been described more clearly. I have a bad habit of being terse.